hera_pspec/pspecbeam.py

```diff
@@ -358,30 +358,41 @@ def power_beam_sq_int(self, pol='pI'):
 
 class PSpecBeamUV(PSpecBeamBase):
 
-    def __init__(self, beam_fname, cosmo=None):
+    def __init__(self, uvbeam, cosmo=None):
         """
         Object to store the primary beam for a pspec observation.
         This is subclassed from PSpecBeamBase to take in a pyuvdata
-        UVBeam object.
+        UVBeam filepath or object.
 
         Parameters
         ----------
-        beam_fname: str
-            Path to a pyuvdata UVBeam file.
+        uvbeam: str or UVBeam object
+            Path to a pyuvdata UVBeam file or a UVBeam object.
         
         cosmo : conversions.Cosmo_Conversions object, optional
             Cosmology object. Uses the default cosmology object if not 
             specified. Default: None.
         """
-        self.primary_beam = UVBeam()
-        self.primary_beam.read_beamfits(beam_fname)
+        # setup uvbeam object
+        if isinstance(uvbeam, str):
+            uvb = UVBeam()
+            uvb.read_beamfits(uvbeam)
+        else:
+            uvb = uvbeam
 
-        self.beam_freqs = self.primary_beam.freq_array[0]
+        # get frequencies and set cosmology
+        self.beam_freqs = uvb.freq_array[0]
         if cosmo is not None:
             self.cosmo = cosmo
         else:
             self.cosmo = conversions.Cosmo_Conversions()
 
+        # setup primary power beam
+        self.primary_beam = uvb
+        if uvb.beam_type == 'efield':
+            self.primary_beam.efield_to_power(inplace=True)
```

Comment:
efield_to_power is called for linear polarizations, for pseudo-stokes you need to use the function efield_to_pstokes.

Reply:
So this is done assuming one only wants linear dipole polarization. This function is (purposely) _not_ trying to be smart and _will not_ convert the efield to pstokes. To `pspecbeam` for pstokes parameters, one needs to feed a pstokes power beam. (see the unit tests added where I test for exactly this case). I'll make a comment specifying this in the docstring!